libyul/optimiser/LoadResolver.cpp

```diff
@@ -38,21 +49,39 @@ void LoadResolver::run(OptimiserStepContext& _context, Block& _ast)
 	LoadResolver{
 		_context.dialect,
 		SideEffectsPropagator::sideEffects(_context.dialect, CallGraphGenerator::callGraph(_ast)),
-		!containsMSize
+		!containsMSize,
+		_context.expectedExecutionsPerDeployment
 	}(_ast);
 }
 
 void LoadResolver::visit(Expression& _e)
 {
 	DataFlowAnalyzer::visit(_e);
 
+	auto isKeccak = [&](YulString const& _name) -> bool
```

Comment:
Can you use the same mechanism that we are already using for mload, i.e. define a function in the dialect and then compare by name?

Reply:
Sure, I'll implement this.

Reply:
I actually implemented this. But I think it ends up being an overkill. I would just prefer the lambda.